clash-lib/src/Clash/Driver/Types.hs

```diff
@@ -540,6 +536,14 @@ newtype SdcInfo = SdcInfo
   { sdcClock :: [(Text, VDomainConfiguration)]
   }
 
+data ClashEnv = ClashEnv
+  { envOpts        :: ClashOpts
+  , envTyConMap    :: TyConMap
+  , envTupleTyCons :: IntMap TyConName
+  , envPrimitives  :: CompiledPrimMap
+  , envCustomReprs :: CustomReprs
+  } deriving (Generic, NFData)
+
```

Comment:
Why did this move down here? You don't do anything with it after which would need the new lenses in scope

Reply:
`TemplateHaskell` :sparkles: